src/WebJobs.Script.Grpc/Server/GrpcServer.cs

```diff
@@ -2,70 +2,61 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
-using System.Linq;
-using System.Threading;
 using System.Threading.Tasks;
-using Grpc.Core;
-using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
+using Microsoft.Azure.WebJobs.Script.Eventing;
+using Microsoft.Azure.WebJobs.Script.Workers;
 using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.Extensions.Hosting;
 
 namespace Microsoft.Azure.WebJobs.Script.Grpc
 {
-    public class GrpcServer : IRpcServer, IDisposable
+    public class GrpcServer : IRpcServer, IDisposable, IAsyncDisposable
     {
-        private int _shutdown = 0;
-        private Server _server;
+        private readonly IHostBuilder _grpcHostBuilder;
+        private readonly int _port;
         private bool _disposed = false;
-        public const int MaxMessageLengthBytes = int.MaxValue;
+        private IHost _grpcHost;
 
-        public GrpcServer(FunctionRpc.FunctionRpcBase serviceImpl)
+        public GrpcServer(IScriptEventManager scriptEventManager)
         {
-            ChannelOption maxReceiveMessageLength = new ChannelOption(ChannelOptions.MaxReceiveMessageLength, MaxMessageLengthBytes);
-            ChannelOption maxSendMessageLength = new ChannelOption(ChannelOptions.MaxSendMessageLength, MaxMessageLengthBytes);
-            ChannelOption[] grpcChannelOptions = { maxReceiveMessageLength, maxSendMessageLength };
-            _server = new Server(grpcChannelOptions)
-            {
-                Services = { FunctionRpc.BindService(serviceImpl) },
-                Ports = { new ServerPort("127.0.0.1", ServerPort.PickUnused, ServerCredentials.Insecure) }
-            };
+            _port = WorkerUtilities.GetUnusedTcpPort();
```

Comment:
I didn't find a reliable way for us to get a port and record that without significant changes. Kestrel has a way to declare that it can pick up any unused port, but for us to be keep track of that becomes more involved, and I suspect there could be races if we need the port number before the server started.

Is there a problem reusing that helper?

Reply:
Just making sure we were utilizing sth built in. This is OK!